app/models/base_input.rb

```diff
@@ -0,0 +1,39 @@
+class BaseInput
```

Comment:
I don't feel convinced yet that this is a better approach. I tried it, but I saw that there were some conflicts with the `self.<method>` methods. Probably for that to work I have to turn it into a `Concern`?